cmd/generate_repo_config/generate_repo_config.go

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2016 The Bazel Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+
+	"github.com/bazelbuild/bazel-gazelle/repo"
+	"github.com/bazelbuild/bazel-gazelle/rule"
+)
+
+var (
+	configSource = flag.String("config_source", "", "a file that is read to learn about external repositories")
+	dest         = flag.String("dest", "", "destination directory")
+)
+
+var repoKind = map[string]rule.KindInfo{
+	"go_repository": {
+		MatchAttrs:    []string{"importpath"},
+		NonEmptyAttrs: map[string]bool{"importpath": true},
+	},
+}
+
+func main() {
```

Comment:
Could we do this in Starlark instead of Go? `_find_macro_file_labels` in `go_repository.bzl` is close to what we need already.